src/Altinn.Apps/AppTemplates/AspNet/App.IntegrationTests/EndToEndTests/NabovarselEndToEnd.cs

```diff
@@ -0,0 +1,279 @@
+            httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, $"{instancePath}/validate");
+            response = await client.SendAsync(httpRequestMessage);
+            responseContent = await response.Content.ReadAsStringAsync();
+            List<ValidationIssue> messages = (List<ValidationIssue>)JsonConvert.DeserializeObject(responseContent, typeof(List<ValidationIssue>));
+            Assert.Empty(messages);
+            #endregion
+
+            // TODO. Add verification of not able to update message and check that statues is updated
+            #region push to next step
+            httpRequestMessage = new HttpRequestMessage(HttpMethod.Put, $"{instancePath}/process/next");
+            response = await client.SendAsync(httpRequestMessage);
+            responseContent = await response.Content.ReadAsStringAsync();
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            #endregion
+
+            #region Get Status after next
+            httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, $"{instancePath}/process");
+            response = await client.SendAsync(httpRequestMessage);
+            responseContent = await response.Content.ReadAsStringAsync();
+            processState = (ProcessState)JsonConvert.DeserializeObject(responseContent, typeof(ProcessState));
+            Assert.Equal("Task_2", processState.CurrentTask.ElementId);
+            #endregion
+
+            #region Get Form DataElement
+
+            dataType = application.DataTypes.FirstOrDefault(r => r.TaskId != null && r.TaskId.Equals(processState.CurrentTask.ElementId));
+
+            DataElement dataElementForm = instance.Data.FirstOrDefault(r => r.DataType.Equals(dataType.Id));
+
+            httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, instancePath + "/data/" + dataElementForm.Id)
+            {
+            };
+
+            response = await client.SendAsync(httpRequestMessage);
+            responseContent = await response.Content.ReadAsStringAsync();
+            SvarPaaNabovarselType skjema = (SvarPaaNabovarselType)JsonConvert.DeserializeObject(responseContent, typeof(SvarPaaNabovarselType));
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            #endregion
+            #region Update Form DataElement
+            string requestJson = JsonConvert.SerializeObject(skjema);
+            StringContent httpContent = new StringContent(requestJson, Encoding.UTF8, "application/json");
+
+            httpRequestMessage = new HttpRequestMessage(HttpMethod.Put, instancePath + "/data/" + dataElementForm.Id)
+            {
+                Content = httpContent
+            };
+            response = await client.SendAsync(httpRequestMessage);
+            responseContent = await response.Content.ReadAsStringAsync();
+            Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+            #endregion
+            #region push to next step
```

Comment:
I'm assuming conflict is due to the form data not being valid? But a short comment on why conflict is expected would be useful.

Reply:
updated comments